packages/reactive-element/src/decorators/custom-element.ts

```diff
@@ -13,6 +13,125 @@
 
 import type {Constructor} from './base.js';
 
+declare global {
+  interface CustomElementRegistry {
+    getName: (ctor: CustomElementConstructor) => string | null;
+  }
+}
+
+export type DefineErrorCallback = (
+  error: Error,
+  tagName: string,
+  ctor: CustomElementConstructor
+) => void;
+
+/**
+ *
+ * @param tagName The tag name of the custom element to define.
+ * @param ctor The constructor of the custom element to define
+ * @param onError An optional function which can handle a define exception
+ * error. If provided, the exception is caught.
+ * @param onError
+ * @returns void
+ */
+export const define = (
+  tagName: string,
+  ctor: CustomElementConstructor,
+  onError?: DefineErrorCallback
+) => {
+  if (customElements.getName?.(ctor) === tagName) {
```

Comment:
This should be the same and has wider support:

```suggestion
  if (customElements.get(tagName) === ctor) {
```